### PR TITLE
Ch 855

### PR DESCRIPTION
**Pull request by alanevans**
Making this PR against CH-845 as that was its starting point.

This adds some limited support for personalizable fields on field collections on other entities. 

The main limitation is that only one level of field collection is supported - nested field collections are not handled here.

### CODE REVIEW

File: modules/personalize_fields/personalize_fields.module [hunk 1]
```diff
@@ -326,87 +431,141 @@ function personalize_fields_form_submit($form, &$form_state) {
       $form_state[$entity_type]->option_sets = array();
     }
     foreach($values['personalized_fields'] as $key => $field) {
-      // Attempt to load an Option Set that's already been saved in the database.
```
Comment (alanevans) [inline modules/personalize_fields/personalize_fields.module:329]: This big red chunk has all moved into personalize_fields_process_form_option_sets(), see below.

File: modules/personalize_fields/personalize_fields.module [hunk 2]
```diff
@@ -8,6 +8,13 @@
 define('PERSONALIZE_FIELDS_CACHE_BIN', 'cache_personalize_fields');
 
 /**
```
Comment (alanevans) [inline modules/personalize_fields/personalize_fields.module:10]: The following chunk is for CH-895.

File: modules/personalize_fields/personalize_fields.module [hunk 3]
```diff
@@ -326,87 +431,141 @@ function personalize_fields_form_submit($form, &$form_state) {
...
+    $option_set = new StdClass();
+    $options = array();
+    foreach($loaded_option_set as $prop => $value) {
+      if (is_array($value)) {
+        $option_set->{$prop} = array();
+        foreach($value as $i => $array_value) {
+          if ($prop === 'options') {
+            $options[$array_value['option_id']] = $i;
           }
+          $option_set->{$prop}[$i] = $array_value;
```
Comment (alanevans) [inline modules/personalize_fields/personalize_fields.module:524]: This is the one line that changed behaviour from the cut-n-paste chunk above: see CH-899 for background.

File: modules/personalize_fields/personalize_fields.module [hunk 4]
```diff
@@ -326,87 +431,141 @@ function personalize_fields_form_submit($form, &$form_state) {
...
-      // it will be processed on hook_entity_insert.
-      if ($entity_id && isset($option_set->is_new)) {
-        personalize_fields_option_set_save($option_set, $entity_type, $entity, $key);
+  $option_hash = array();
+  foreach ($option_set->options as $delta => $option) {
+    $option_hash[$option['option_id']] = $delta;
+  }
+  $keep_option_deltas = array();
+  // Now update the options based on what the user passed in
+  foreach ($values[$field_name][LANGUAGE_NONE] as $delta => $option) {
```
Comment (alanevans) [inline modules/personalize_fields/personalize_fields.module:539]: These hardcoded LANGUAGE_NONEs didn't look right to me, but I didn't want to go poking around in stuff that is working for now.

Comment (katbailey) [inline modules/personalize_fields/personalize_fields.module:539]: Yes, every time I see them in the code I cringe too - I know that this is going to bite us and it's only a matter of time, but definitely beyond the scope of this story.

**Pull request closed without merging**